### PR TITLE
Migrate to go-libp2p-core

### PR DESCRIPTION
**Pull request by ansermino**
## Changes
- Introduces `github.com/libp2p/go-libp2p-core`
- Removes deprecated libp2p dependencies
- Due to issues faced with build tags Badger has been updated to 0.1.6 which includes minor API changes

## Tests:
```
go test ./...
```

### Fixes: #158

### CODE REVIEW

**Comment by noot** [comment]
having badger issues :'( did you find a fix for this yet?

**Comment by dutterbutter** [comment]
@noot can you post your error output.

**Comment by noot** [comment]
@dutterbutter  https://github.com/ChainSafe/gossamer/issues/166

**Comment by ansermino** [comment]
Managed to fix locally, having issues reproducing in the CI. See link in #166

File: .travis.yml [hunk 1]
```diff
@@ -8,10 +8,10 @@ env:
...
     - CC_TEST_REPORTER_ID=abdad8b2e2ec0bfdd7af57d955c44e9470f4d174e744b824047f3037800f5b40
 
-cache:
-  directories:
-    - $HOME/.cache/go-build
-    - $GOPATH/pkg/mod
+#cache:
+#  directories:
+#    - $HOME/.cache/go-build
+#    - $GOPATH/pkg/mod
```
Comment (dutterbutter) [inline .travis.yml]: What's the reasoning for removing the caching on the builds? Is this just temporary?

Comment (ansermino) [reply .travis.yml]: removed

**Pull request merged**